apps/src/block_utils.js

```diff
@@ -1085,13 +1085,13 @@ exports.createJsWrapperBlockCreator = function (
         // automatically based on a block's returnType (e.g. yellow for "Location").
         if (color) {
           Blockly.cdoUtils.setHSV(this, ...color);
-        } else if (!returnType) {
+        } else if (!returnType && !this.getStyleName()) {
```

Comment:
What's the difference between checking `style` and `this.getStyleName()`?

I'm just curious why the logic you changed here and the logic related to styles above didn't get combined into something like...

```
const style = this.getStyleName();
if (style) {
  this.setStyle(style);
} else if (color) {
  Blockly.cdoUtils.setHSV(...);
} else {
  Blockly.cdoUtils.setHSV(this, ...DEFAULT_COLOR);
  this.setStyle('default');
}

Reply:
This is the joy of `block_utils.js` madness. In this scope, `style` refers to the value set in the block config JSON. Above (https://github.com/code-dot-org/code-dot-org/commit/53e3d31e00c44814cdab0cd60affe59583e0a18d), we use this value to actually set the style on the block:

```
        if (style) {
          // Google Blockly method. No-op for CDO Blockly.
          this.setStyle(style);
        }
```

`setStyle` does nothing for CDO Blockly (e.g. Minecraft). Similiarly, `getStyleName` returns `undefiend` for CDO Blockly.  The line you tagged is using `this.getStyleName()` to check whether a block actually has a style. This will be falsey if `style` is missing OR if we're in CDO Blockly. In either situation, it's safe to apply the default color.